control_toolbox/include/control_toolbox/pid.hpp

```diff
@@ -39,64 +39,151 @@
 #include <limits>
 #include <string>
 
+#include "fmt/format.h"
 #include "rclcpp/duration.hpp"
 #include "realtime_tools/realtime_buffer.hpp"
 
 namespace control_toolbox
 {
-class AntiwindupStrategy
+/**
+ * \brief Antiwindup strategy for PID controllers.
+ *
+ * This class defines various antiwindup strategies that can be used in PID controllers.
+ * It allows setting the type of antiwindup strategy and validates the parameters accordingly.
+ *
+ * \param i_max Upper integral clamp.
+ * \param i_min Lower integral clamp.
+ * \param u_max Upper output clamp.
+ * \param u_min Lower output clamp.
+ * \param tracking_time_constant Specifies the tracking time constant for the 'back_calculation' strategy. If set
+ *    to 0.0 when this strategy is selected, a recommended default value will be applied.
+ * \param legacy_antiwindup Anti-windup functionality. When set to true, limits
+      the integral error to prevent windup; otherwise, constrains the
+      integral contribution to the control output. i_max and
+      i_min are applied in both scenarios.
+  * \param error_deadband Error deadband is used to avoid integration when the error is within a small range.
```

Comment:
```suggestion
  * \param error_deadband Error deadband is used to avoid integration when the error is within a small range.
```
```suggestion
  * \param error_deadband Error deadband is used to stop integration when the error is within the given range.
```